tests/integration/test_bundle.py

```diff
@@ -137,3 +203,144 @@ async def test_multiple_kratos_external_idp_integrators(ops_test: OpsTest):
 
     action_output = await get_redirect_uri_action.wait()
     assert "redirect-uri" in action_output.results
+
+
+async def test_kratos_scale_up(ops_test: OpsTest) -> None:
+    """Check that kratos works after it is scaled up."""
+    app_name = "kratos"
+
+    app = ops_test.model.applications[app_name]
+
+    await app.scale(3)
+
+    await ops_test.model.wait_for_idle(
+        raise_on_blocked=False,
+        raise_on_error=False,
+        status="active",
+        timeout=2000,
+    )
+
+    admin_address = await get_app_address(ops_test, TRAEFIK_ADMIN_APP)
+    health_check_url = (
+        f"https://{admin_address}/{ops_test.model.name}-{app_name}/admin/health/ready"
+    )
+    resp = requests.get(health_check_url, verify=False)
```

Comment:
I would like to learn:

- Does making this additional health check request guarantee to tell us all units are ready or healthy?

Reply:
Not really, we can't tell to which unit you will be redirected to. TBH this check does not offer much, as we already know that the units are in an active state, other than verifying that ONLY one of the units is still accessible. 

I have no problem with removing it or changing it if you have a better idea

Reply:
ic~ thanks for the clarifications. If it does not help tell us whether the scaling is good or not, and the above `ops_test.model.wait_for_idle` already lets us know the scaled-up units reach a steady state, I think we could remove it.

Reply:
Sorry, I would like to learn:

Does the `OpsTest` offers some APIs like checking the number of units and the states of units at the current moment? I guess this is what the test probably would like to verify at the end.

Reply:
If you want to wait for a specific number of units, you can use the [wait_for_units param](https://github.com/juju/python-libjuju/blob/ad62156b3bd18c949cd7f65a6b528cbaf023e121/juju/model.py#L2613). But in our case, we just want all the units to become active, which means that if `wait_for_idle` returns then we are fine (no need to specify `wait_for_units`).